pkg/kmsplugin/kms.go

```diff
@@ -92,13 +92,15 @@ func ParseError(err error) (errorType KMSErrorType) {
 
 	// AWS SDK Go for KMS does not "yet" define specific error code for a case where a customer specifies the deleted key
 	// "AccessDeniedException" error code may be returned when (1) CMK does not exist (not pending delete),
-	// or (2) corresponding IAM role is not allowed to access the key.
-	// Thus we only want to mark "AccessDeniedException" as user-induced for the case (1).
+	// or (2) user explicitly denied access to the key via resource policy,
+	// or (3) corresponding IAM role is not allowed to access the key.
+	// Thus we only want to mark "AccessDeniedException" as user-induced for the case (1) and (2).
 	// e.g., "AccessDeniedException: The ciphertext refers to a customer master key that does not exist, does not exist in this region, or you are not allowed to access."
 	//       or "AccessDeniedException: User xxx is not authorized to perform: xxx on this resource because the resource does not exist in this Region, no resource-based policies allow access, or a resource-based policy explicitly denies access"
+	//       or "AccessDeniedException: User xxx is not authorized to perform: xxx on this resource with an explicit deny in a resource control policy"
 	// KMS service may change the error message, so we do the string match.
 	case "AccessDeniedException":
-		if strings.Contains(ae.ErrorMessage(), "does not exist") {
+		if strings.Contains(ae.ErrorMessage(), "does not exist") || strings.Contains(ae.ErrorMessage(), "explicit deny in a resource control policy") {
```

Comment:
what if AOK is disable due to whatever reason?

Reply:
We have alarms for AOK disabling or throttling, that's why we also exclude throttling from livez regardless of the key type https://github.com/kubernetes-sigs/aws-encryption-provider/blob/master/pkg/plugin/plugin_v2.go#L119